### PR TITLE
Update karpenter to v0.18.0 and use production-ready image

### PR DESCRIPTION
**Pull request by linki**
CRD changed in v0.16.2.

I split them in two separate files like upstream to pull them easier in the future: https://github.com/aws/karpenter/tree/main/charts/karpenter/crds

### CODE REVIEW

**Comment by linki** [comment]
👍

File: cluster/manifests/z-karpenter/deployment.yaml [hunk 1]
```diff
@@ -38,7 +38,7 @@ spec:
         effect: NoSchedule
       containers:
         - name: controller
-          image: "public.ecr.aws/karpenter/controller:v0.16.1@sha256:ed1f97bc2cd67e63b45546955cf763890182f1f97586fca5e6cbd25f175b9c56"
+          image: "container-registry.zalando.net/teapot/karpenter:v0.18.0-main-3"
```
Comment (mikkeloscar) [inline cluster/manifests/z-karpenter/deployment.yaml]: The webhook image should also be changed. It's in the same pod template further down on line 83

Comment (linki) [reply cluster/manifests/z-karpenter/deployment.yaml]: 🤦  Thanks!

**Comment by linki** [comment]
👍

**Comment by myaser** [comment]
👍🏻

**Pull request merged**